sled-agent/src/services.rs

```diff
@@ -4272,59 +4283,53 @@ impl ServiceManager {
                 );
                 *request = new_request;
 
+                // Add SMF properties here and restart zone-network-setup service
                 let first_address = request.addresses.get(0);
                 let address = first_address
                     .map(|addr| addr.to_string())
                     .unwrap_or_else(|| "".to_string());
 
-                for addr in &request.addresses {
-                    if *addr == Ipv6Addr::LOCALHOST {
-                        continue;
-                    }
-                    info!(
-                        self.inner.log,
-                        "Ensuring address {} exists",
-                        addr.to_string()
-                    );
-                    let addr_request =
-                        AddressRequest::new_static(IpAddr::V6(*addr), None);
-                    zone.ensure_address(addr_request).await?;
-                    info!(
-                        self.inner.log,
-                        "Ensuring address {} exists - OK",
-                        addr.to_string()
-                    );
-                }
+                // Set new properties for the network set up service and refresh
+                let nw_setup_svc = ZoneNetworkSetupService {};
+                let nsmfh = SmfHelper::new(&zone, &nw_setup_svc);
+
+                nsmfh.delpropvalue("config/gateway", "*")?;
+                nsmfh.delpropvalue("config/static_addr", "*")?;
 
-                // When the request addresses have changed this means the underlay is
-                // available now as well.
                 if let Some(info) = self.inner.sled_info.get() {
-                    info!(
+                    nsmfh.addpropvalue_type(
+                        "config/gateway",
+                        &format!("{}", info.underlay_address),
```

Comment:
This method takes any type that impls `ToString`, so we don't need to format here. We can pass `&info.underlay_address` I think.